### ISSUE TITLE
Implement endpoint: /core/version-check

### ISSUE THREAD

**Issue by chuckadams** [aspirepress/AspireCloud#52]

**Comment by Tmeister** [comment]
Hi all,

The most recent version of this endpoint is located here: `https://api.wordpress.org/core/version-check/1.7/` the versions listed in the endpoint go from `6.6.2` to `4.1.41`

This is the template used for each version:

```json
{
      "response": "upgrade",
      "download": "https://downloads.wordpress.org/release/wordpress-6.6.2.zip",
      "locale": "en_US",
      "packages": {
        "full": "https://downloads.wordpress.org/release/wordpress-6.6.2.zip",
        "no_content": "https://downloads.wordpress.org/release/wordpress-6.6.2-no-content.zip",
        "new_bundled": "https://downloads.wordpress.org/release/wordpress-6.6.2-new-bundled.zip",
        "partial": false,
        "rollback": false
      },
      "current": "6.6.2",
      "version": "6.6.2",
      "php_version": "7.2.24",
      "mysql_version": "5.5.5",
      "new_bundled": "6.4",
      "partial_version": false
    },
```

Questions @chuckadams @sarah-savage :

- Will this information and zip files be part of the sync project?
- Will we store all the old versions and new patches in the future?
- Or is it better to just bypass this endpoint?

**Comment by chuckadams** [comment]
AspireSync might be able to grab the endpoints, but I think for now we can leave this one as pass-through.

Client implementation is in [wp_version_check](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/update.php#L27) and appears to do a POST to `/core/version/check/1.7`.  

* Despite taking a POST, most input comes through the query string, including the versions of every single plugin installed.

* An optional query parameter `channel` is supported and can be one of `beta`, `rc`, `development`, or `branch-development`.  The main channel is accessed by leaving out the parameter.

* POST data is json and has at least 'translations' (itself a json object), but wp_version_check can take arbitrary extra data to post via $extra_stats, which presumably are just reporting and don't affect the response.  We'll want to capture some interactions with this endpoint to make sure.

**Comment by chuckadams** [comment]
I'm having a devil of a time finding the implementation for this in the .org repo.  Adding the help-wanted tag.

**Comment by chuckadams** [comment]
After today's meeting, I decided what we'll do is still pass the request through, but then cache it on our side (normalizing the request to a minimal key) and apply URL rewrites to point to AC's /download endpoint, which already supports core.  

This is basically making AC more of a "smart proxy" instead of the all-or-nothing approach we have now where we either handle an endpoint or completely punt and pass it on unmolested.  In this new hybrid mode, we now molest everything.  _(note: that should probably not go on press releases)_

**Comment by afragen** [comment]
Core rollbacks are a different beast. They are ideally in place if there was an issue during the core download/update process.

**Comment by chuckadams** [comment]
Fully implementing this endpoint is looking less and less likely by launch: we don't have visibility into the translation asset versions, so we would need to defer to upstream for an updated source.  We can of course cache these assets ourselves and avoid upstream requests that way, but that either means expanding AspireSync's duties to pre-fetch them, or it means we roll that functionality into AC.  I'm partial to the latter, but that does entail some design changes that aren't likely to happen in less than a week.

TLDR: I'm actively working on this, but no longer calling it a blocker for launch.
